checkov/terraform/parser2.py

```diff
@@ -0,0 +1,695 @@
+import json
+import logging
+import os
+import re
+from pathlib import Path
+from typing import Mapping, Optional, Dict, Any, List, Callable, Tuple
+
+import deep_merge
+import hcl2
+import jmespath
+
+from checkov.common.runners.base_runner import filter_ignored_directories
+from checkov.common.variables.context import EvaluationContext, VarReference
+from checkov.terraform.module_loading.registry import ModuleLoaderRegistry
+from checkov.terraform.module_loading.registry import module_loader_registry as default_ml_registry
+
+
+LOGGER = logging.getLogger(__name__)
+
+_VAR_PATTERN = re.compile(r'\${([^{}]+?)}')
+_SIMPLE_TYPES = frozenset(["string", "number", "bool"])
+
+_RESOURCE_REF_PATTERN = re.compile(r'[\d\w]+(\.[\d\w]+)+')
+
+
+class Parser2:
```

Comment:
@robeden  
I think it's safe to rename this class to `Parser` now (and the file name as well).

Reply:
Done